### PR TITLE
Concatenate 

### PR DESCRIPTION
**Pull request by optimusprime009**
https://github.com/unifyai/ivy/issues/4183

### CODE REVIEW

**Comment by optimusprime009** [comment]
why my merging is blocked

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hey,

For this PR you should only have changes to files for the relevant task. There seems to be changes in `ivy_tests/test_array_api`. These would need to be removed :)

Thanks!

**Comment by optimusprime009** [comment]
so are we do not need to write the test for the front end function?

**Comment by optimusprime009** [comment]
> 

sorry they were committed by codespace by accident

**Comment by optimusprime009** [comment]
can you connect the issue #4183 to this pull request

**Comment by optimusprime009** [comment]
ivy/functional/frontends/torch/linalg.py:10:89: E501 line too long (92 > 88 characters) what to do about this

**Comment by optimusprime009** [comment]
#4183

**Comment by jkeane508** [comment]
> ivy/functional/frontends/torch/linalg.py:10:89: E501 line too long (92 > 88 characters) what to do about this ?

This is a lint error, Ivy has formatting rules for code, you can install a pre-commit hook to prevent these issues in the future -https://lets-unify.ai/ivy/contributing/setting_up.html#pre-commit

File: .gitmodules
Comment (jkeane508) [inline .gitmodules:1]: We cannot accept PRs with changes to the .gitignore file

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py:1]: this can be removed

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py:5]: This can be removed

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py:7]: This can just be `# concatenate`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py:14]: this can be removed

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_layers.py:15]: this can also be removed

File: ivy/functional/frontends/tensorflow/keras/layers.py
Comment (jkeane508) [inline ivy/functional/frontends/tensorflow/keras/layers.py:1]: we should add `# global` above this line

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hi, in this PR there can't be any changes to files not relevant to the task. This PR deletes the array API folder and makes changes to the .gitignore, we can't merge until these are resolved.

Moreover, we should limit comments where possible. For imports, we should split them into global and local imports.

Thanks

**Comment by optimusprime009** [comment]
> 

how we are going to do that?

**Comment by optimusprime009** [comment]
> Hi, in this PR there can't be any changes to files not relevant to the task. This PR deletes the array API folder and makes changes to the .gitignore, we can't merge until these are resolved.
> 
> Moreover, we should limit comments where possible. For imports, we should split them into global and local imports.
> 
> Thanks

How are we gonna work on that?

**Comment by jkeane508** [comment]
> why my merging is blocked ?

You don't have permission to merge PRs to the master branch of Ivy. The reviewer will merge the PR when its accepted :)

**Comment by jkeane508** [comment]
> > Hi, in this PR there can't be any changes to files not relevant to the task. This PR deletes the array API folder and makes changes to the .gitignore, we can't merge until these are resolved.
> > Moreover, we should limit comments where possible. For imports, we should split them into global and local imports.
> > Thanks
> 
> How are we gonna work on that? you told me to remove the test_Array_api folder.

Hey,

I didn't ask you to remove the folder, I asked you to remove changes to the folder. In your original commit, you had made changes to the code in the array API folder which would have made changes to the master branch. In general, there should only be changes made to the files relevant to the task you are working on :)

Hope this helps

**Comment by optimusprime009** [comment]
but I didn't commit those changes on the test array API it is automatically done by codespace. Do you know what I should do next?

**Comment by jkeane508** [comment]
> but I didn't commit those changes on the test array API it is automatically done by codespace. Do you know what I should do next?

Hi,

In this case, I would rebase your local branch with the Ivy Master branch, such that your local base is identical to the master branch. I would then make implement the function and test locally, and commit only the relevant changes to the task not all changes on the branch :)

**Comment by optimusprime009** [comment]
what should I do now? regarding the files, should I create a new pull request or delete the entire commit of the branch?

**Comment by jkeane508** [comment]
> what should I do now? regarding the files, should I create a new pull request or delete the entire commit of the branch?

You could probably just rebase your branch to the master branch of Ivy :)

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**
